### DISCUSSION TITLE
80. 커버링 인덱스가 무엇이며, 일반 인덱스에 비해 성능이 뛰어난 이유는 무엇인가요?

### DISCUSSION THREAD

**Discussion by cutehumanS2** [jewelry-box/cs-interview#82] [General]
커버링 인덱스가 무엇이며, 일반 인덱스에 비해 성능이 뛰어난 이유는 무엇인가요?

**Comment by linirini** [comment]
커버링 인덱스는 쿼리가 요청하는 모든 데이터를 인덱스만으로 처리할 수 있는 인덱스이다. **테이블의 데이터 행(row)**에 접근하지 않고, 인덱스만으로 결과를 반환할 수 있다.
일반 인덱스와 달리 테이블 데이터 접근을 줄여 디스크 I/O를 최소화하여 성능이 매우 뛰어나다.
일반 인덱스는 쿼리가 요청한 데이터가 인덱스에 없다면, 인덱스를 통해 데이터를 찾은 후 테이블의 레코드를 조회해야하며 이 과정에서 추가적인 I/O 비용이 발생한다.
반면, 커버링 인덱스는 쿼리가 요청한 모든 데이터가 인덱스에 포함되어 있으므로, 테이블 데이터에 접근할 필요가 없어 추가 I/O가 발생하지 않아 성능이 좋다.
